packages/grid/x-data-grid-pro/package.json

```diff
@@ -52,8 +52,8 @@
     "reselect": "^4.1.5"
   },
   "peerDependencies": {
-    "@mui/material": "^5.2.8",
-    "@mui/system": "^5.2.8",
+    "@mui/material": "^5.4.1",
+    "@mui/system": "^5.4.1",
```

Comment:
This might be considered a breaking change, but I would argue that the breaking change was already released in https://github.com/mui/mui-x/issues/2998#issuecomment-1036337411 so we might as well be upfront with it.